### PR TITLE
Check footer, hero and loading indicator against pr checklist

### PR DESCRIPTION
**Pull request by GCHQ-Developer-530**
## Summary of the changes
Update hero and footer to properly allow for slots instead of props. Update stories to have correct props.

## Related issue
Part of #1817 

### General 

- [x] Changes to docs package checked and committed.
- [x] All acceptance criteria reviewed and met. 

### Testing

- [x] Relevant unit tests and visual regression tests added. 
- [x] Visual testing against Figma component specification completed. 

### Accessibility 

- [x] Accessibility Insights FastPass performed.
- [x] A11y unit test added and yields no issues.
- [x] A11y plug-in on Storybook yields no issues. 
- [x] Manual screen reader testing performed using NVDA and VoiceOver. 
- [x] Manual keyboard testing for keyboard controls and logical focus order. 
- [x] Correct roles used and ARIA attributes used correctly where required. 
- [x] Logical heading structure is maintained, and the HTML elements used for headings can be changed to fit within the wider page structure. 

### Resize/zoom behaviour 

- [x] Page can be zoomed to 400% with no loss of content. 
- [x] Screen magnifier used with no issues. 
- [x] Text resized to 200% with no loss of content.
- [x] Text spacing increased as per the [WCAG 1.4.12 success criterion](https://www.w3.org/TR/WCAG21/#text-spacing) with no loss of content.

### System modes

- [x] Browser setting 'prefers reduced motion' tested. No animations or motion visible whilst this setting is on. 
- [x] Windows High Contrast mode tested with no loss of content.
- [x] System light and dark mode tested with no loss of content.
- [x] Browser support tested (Chrome, Safari, Firefox and Edge). 

### Testing content extremes

- [x] Min/max content examples tested with no loss of content or overflow. 
- [x] All prop combinations work without issue. 
- [x] Tested for FOUC (Flash of Unstyled Content) in both SSR (Server-Side Rendering) and SSG (Static Site Generation) settings.
- [x] Controlled and uncontrolled input components tested.

### CODE REVIEW

File: packages/react/src/component-tests/IcFooter/IcFooter.cy.tsx [hunk 1]
```diff
@@ -517,5 +521,26 @@ describe("IcFooter", () => {
...
+            All content is available under the Open Government Licence v3.0,
+            except source code and code examples which are available under the
+            MIT Licence.
+          </IcTypography>
+        </IcFooter>
+      );
+      cy.checkA11yWithWait();
+      cy.compareSnapshot({
+        name: "slottedDescriptionCaption",
+        testThreshold: setThresholdBasedOnEnv(DEFAULT_TEST_THRESHOLD + 0.05),
```
Comment (GCHQ-Developer-299) [inline packages/react/src/component-tests/IcFooter/IcFooter.cy.tsx:542]: Should there be a new screenshot in the changed files?

**Review by GCHQ-Developer-299 — DISMISSED** [review]

File: packages/react/src/component-tests/IcFooter/IcFooter.cy.tsx [hunk 2]
```diff
@@ -13,11 +13,12 @@ import {
 import { mount } from "cypress/react";
 import { SlottedSVG } from "../../react-component-lib/slottedSVG";
 import { BE_VISIBLE, HAVE_LENGTH } from "../utils/constants";
+import { setThresholdBasedOnEnv } from "../../../cypress/utils/helpers";
 
 const DEFAULT_TEST_THRESHOLD = 0.05;
 
 describe("IcFooter", () => {
-  it(" IcFooter ", () => {
+  it("renders", () => {
```
Comment (GCHQ-Developer-465) [inline packages/react/src/component-tests/IcFooter/IcFooter.cy.tsx:21]:
```suggestion
  it("should render", () => {
```

File: packages/react/src/component-tests/IcFooter/IcFooter.cy.tsx [hunk 3]
```diff
@@ -480,9 +481,12 @@ describe("IcFooter", () => {
       });
     });
 
-    it("footer with no caption", () => {
+    it("footer with no caption or copyright", () => {
```
Comment (GCHQ-Developer-465) [inline packages/react/src/component-tests/IcFooter/IcFooter.cy.tsx:484]:
```suggestion
    it("should render footer with no caption or copyright", () => {
```

File: packages/react/src/component-tests/IcFooter/IcFooter.cy.tsx [hunk 4]
```diff
@@ -517,5 +521,26 @@ describe("IcFooter", () => {
         testThreshold: DEFAULT_TEST_THRESHOLD + 0.05,
       });
     });
+
+    it("render a footer with slotted description and caption", () => {
```
Comment (GCHQ-Developer-465) [inline packages/react/src/component-tests/IcFooter/IcFooter.cy.tsx:525]:
```suggestion
    it("should render a footer with slotted description and caption", () => {
```

File: packages/react/src/component-tests/IcHero/IcHero.cy.tsx [hunk 5]
```diff
@@ -18,7 +25,7 @@ describe("IcHero e2e, A11y and visual regression tests", () => {
     cy.task("generateReport");
   });
 
-  it("IcHero", () => {
+  it("renders", () => {
```
Comment (GCHQ-Developer-465) [inline packages/react/src/component-tests/IcHero/IcHero.cy.tsx:28]:
```suggestion
  it("should render", () => {
```

File: packages/react/src/component-tests/IcHero/IcHero.cy.tsx [hunk 6]
```diff
@@ -40,6 +47,29 @@ describe("IcHero e2e, A11y and visual regression tests", () => {
     cy.checkA11yWithWait();
   });
 
+  it("renders with slotted heading and subheading", () => {
```
Comment (GCHQ-Developer-465) [inline packages/react/src/component-tests/IcHero/IcHero.cy.tsx:50]:
```suggestion
  it("should render with slotted heading and subheading", () => {
```

**Review by GCHQ-Developer-465 — APPROVED** [review]

**Review by GCHQ-Developer-465 — APPROVED** [review]

**Review by GCHQ-Developer-299 — APPROVED** [review]

**Pull request merged**